dscache/dscache.go

```diff
@@ -18,28 +22,48 @@ var (
 
 // Dscache represents an in-memory serialized dscache flatbuffer
 type Dscache struct {
+	Filename            string
 	Root                *dscachefb.Dscache
 	Buffer              []byte
 	ProfileIDToUsername map[string]string
 }
 
-// LoadDscacheFromFile will load a dscache from the given filename
-func LoadDscacheFromFile(filename string) (*Dscache, error) {
-	buffer, err := ioutil.ReadFile(filename)
+// NewDscache will construct a dscache from the given file, or will construct an empty dscache
+// that will save to the given filename
+func NewDscache(fsys qfs.Filesystem, filename string) *Dscache {
```

Comment:
Can we get a comment mentioning that passing the empty string as `filename` will skip loading and saving entirely? It's a great convention, I'd like consumers to know it's both available & intentional.